frontend/src/hooks/connections.ts

```diff
@@ -56,7 +56,12 @@ export function useGetConnection(id?: string) {
 }
 
 const defaultCreateErrorHandler = (err: Error) => {
-  if (!isAxiosError(err)) {
+  if (isAxiosError(err) && err.response?.status === 404) {
```

Comment:
When would it return a 404? If `sample_name` is not one of the samples we have, it'll be a 422